apps/server/src/shared/repo/base.repo.ts

```diff
@@ -12,6 +12,10 @@ export abstract class BaseRepo<T extends BaseEntity> {
 
 	abstract get entityName(): EntityName<T>;
 
+	create(entity: T): T {
```

Comment:
Why do we need this method? Could we just use the entity constructor and then persitst the entity with `save()`?
Also why is there a dedicated implementation in the subclass SchoolRepo?
https://github.com/hpi-schul-cloud/schulcloud-server/blob/f3b18cb4743f94731b61ed482f1ceaadeefb5d75/apps/server/src/shared/repo/school/school.repo.ts#L11

Reply:
Long term, no usecase/module should handle raw entites, only DomainObjects. This is a first step towards this architecture for the older repos. It will make porting them over to DO repos easier. 

The SchoolRepo point is valid though imo. We will fix that

Reply:
The SchoolRepo has this method because we need the generated id of the entity, which we can't get from a persist call.

Reply:
So you are using `em.create()` for assigning the id to the entity? I think that can also be done differently.

Reply:
em.create() is the safest method to create new entities. It ensures proper generation of the entity, including it's ID, which is the important part. Using any other method could have potential problems and create inconsistencies in the database or even duplicate IDs. All Entity CRUD operations should be handled by the EM, everything else is potentially unsafe